packages/calling-component-bindings/src/baseSelectors.ts

```diff
@@ -86,6 +86,17 @@ export const getRemoteParticipants = (
       [keys: string]: RemoteParticipantState;
     }
   | undefined => {
+  /* @conditional-compile-remove(rooms) */
+  // Filter out consumers from remote participants
+  {
+    const remoteParticipants = { ...state.calls[props.callId]?.remoteParticipants };
+    Object.keys(remoteParticipants).forEach((k) => {
+      if (remoteParticipants[k].role === 'Consumer') {
+        delete remoteParticipants[k];
+      }
+    });
+    return remoteParticipants;
```

Comment:
My understanding is these baseSelectors should avoid returning new objects. Selectors build of these using `reselect.createSelector`. AFAIK the createSelector function is memoized based on the inputs. Meaning that when the input are not changed, it returns a memoized result (meaning react also receives a memoized result).
However this change will be creating a new remoteParticipants object every time its called, meaning this will bust the memoization of all the createSelectors using this.

Instead create a new selector called getFilteredRemoteParticipants or something that is a selector on top of this base selector. And have all the relevant sub-selectors use the new selector instead
```ts
export const getFilteredRemoteParticipants = reselect.createSelector(
  [getRemoteParticipants],
  (remoteParticipants) => {
    return remoteParticipants.filter((remoteParticipants) => remoteParticipants.role !=== 'Consumer');
  }
);
```

Let me know if I'm misunderstanding the base selector usage and how things get memoized!
cc @PorterNan if he can confirm.

Reply:
Thanks for this tip about `reselect.createSelector`! I read up a bit about it and you are right that it only returns a new object when one of the values change. 
![image](https://github.com/Azure/communication-ui-library/assets/79475487/bc2ece8b-ae5e-4971-b62d-3c63eb4eae63)
[source](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
Updated.